src/components/app/App.jsx

```diff
@@ -3,9 +3,43 @@ import Form from '../Form'
 import Home from '../Home'
 import TaskList from '../TaskList'
 import './App.css'
+import { useState, useEffect } from 'react'
 
 function App() {
 
+  const [mode, setMode] = useState((localStorage.getItem("modeState") === "true")); // localStorage stores everything as strings hence why true is in string data type
```

Comment:
We can try using `useRef` instead.

```suggestion
const currentMode = useRef(localStorage.getItem("modeState"))
  const [mode, setMode] = useState(currentMode.current === "true"); // localStorage stores everything as strings hence why true is in string data type
```